R/net.mod.simnet.R

```diff
@@ -110,19 +110,39 @@ resim_nets <- function(dat, at) {
   if (tergmLite == TRUE & resimulate.network == TRUE) {
     isTERGM <- ifelse(nwparam$coef.diss$duration > 1, TRUE, FALSE)
     dat <- tergmLite::updateModelTermInputs(dat)
-
-   if (isTERGM == TRUE) {
-     dat$el[[1]] <- tergmLite::simulate_network(p = dat$p[[1]],
-                                                el = dat$el[[1]],
-                                                coef.form = nwparam$coef.form,
-                                                coef.diss =
-                                                   nwparam$coef.diss$coef.adj,
-                                                save.changes = TRUE)
+    
+    if (isTERGM == TRUE) {
+      rv <- tergmLite::simulate_network(state = dat$p[[1]]$state,
+                                        coef = c(nwparam$coef.form,
+                                                 nwparam$coef.diss$coef.adj),
+                                        control = dat$control$mcmc.control[[1]],
+                                        save.changes = TRUE)
+
+      dat$el[[1]] <- rv$el
+
+      if (dat$control$tergmLite.track.duration) {
+        dat$p[[1]]$state$nw0 %n% "time" <- rv$state$nw0 %n% "time"
+        dat$p[[1]]$state$nw0 %n% "lasttoggle" <- rv$state$nw0 %n% "lasttoggle"
+      }
     } else {
-      dat$el[[1]] <- tergmLite::simulate_ergm(p = dat$p[[1]],
-                                              el = dat$el[[1]],
-                                              coef = nwparam$coef.form)
+      rv <- tergmLite::simulate_ergm(state = dat$p[[1]]$state,
+                                     coef = nwparam$coef.form,
+                                     control = dat$control$mcmc.control[[1]])
+
+      dat$el[[1]] <- rv$el
     }
+    
+    if (dat$control$save.nwstats == TRUE) {
+      nwL <- networkLite(dat$el[[1]], dat$attr)
+      if (dat$control$tergmLite.track.duration) {
+        nwL %n% "time" <- dat$p[[1]]$state$nw0 %n% "time"
+        nwL %n% "lasttoggle" <- dat$p[[1]]$state$nw0 %n% "lasttoggle"
+      }
+      dat$stats$nwstats[[1]] <- rbind(dat$stats$nwstats[[1]],
+                                        summary(dat$control$nwstats.formulas[[1]], 
```

Comment:
just a note that it is `nwstats.formula` (singular), and not an embedded list (so control$nwstats.formula would be of class formula not list). I think we had decided yesterday to keep it like that for now.

Reply:
see here: https://github.com/statnet/EpiModel/issues/541#issuecomment-856119470

Reply:
init_tergmLite organizes some things into lists in anticipation of the general multiple-network case

the user of core EpiModel would specify `nwstats.formula = whatever`, as they do for EpiModel tergm simulation, and not need to worry about these internal details